src/utils/rates.js

```diff
@@ -20,28 +20,49 @@
 import { BigNumber } from 'bignumber.js';
 
 // Constants
-import { ETH, USD } from 'constants/assetsConstants';
+import { ADDRESS_ZERO, USD } from 'constants/assetsConstants';
 
 // Utils
 import { wrapBigNumber } from 'utils/bigNumber';
+import { valueForAddress } from 'utils/common';
 
 // Types
-import type { RatesBySymbol, Currency } from 'models/Rates';
+import type { Currency, RatesByAssetAddress } from 'models/Rates';
 
-export function getUsdToFiatRate(rates: RatesBySymbol, currency: Currency): ?number {
+
+export const getUsdToFiatRate = (rates: RatesByAssetAddress, currency: Currency): ?number => {
```

Comment:
Looks somewhat weird that we would pass rates from different chain here ;-)

Reply:
Wdym?

Reply:
We are doing EUR/USD or GBP/USD and we base the calculation on different native token prices depending on which chain rates we use. Hopefully all chains will result it the same EUR/USD rate, but probably mainnet would be most reliable one.

Seems like this function is not used a lot, maybe we could inline it into `usdToFiatRateSelector` to make sure it's always based on mainnet, wdyt?

Reply:
Will see what we can do here.